src/lib/components/Swap/SwapButton.tsx

```diff
@@ -167,17 +172,52 @@ export default function SwapButton({ disabled }: SwapButtonProps) {
       })
   }, [addTransaction, inputCurrencyAmount, outputCurrencyAmount, setDisplayTxHash, swapCallback, tradeType])
 
+  const ButtonText = useCallback(() => {
```

Comment:
Never seen a useCallback used to make a react component. Why this over `useMemo`?

Reply:
useMemo gives you a value, useCallback gives you a function. You need the function call for the component constructor. 

I've **also** never seen useCallback used this way, but it... feels right? I usually see this pattern used w/o the hook to define a locally scoped function component, but I think that's strictly worse, right?


<img width="688" alt="image" src="https://user-images.githubusercontent.com/5773490/155634357-fcb14103-67c2-49bb-a55e-75ccf4a21998.png">